src/System.Runtime/tests/System/Reflection/InvokeRefReturn.netcoreapp.cs

```diff
@@ -75,6 +75,26 @@ public static unsafe void TestNullRefReturnOfPointer()
             Assert.NotNull(p);
             Assert.Throws<NullReferenceException>(() => p.GetValue(tc));
         }
+        
+        [Fact]
+        [ActiveIssue("TFS 603305 - Bring Project N up to sync", TargetFrameworkMonikers.UapAot)]
+        public static unsafe void TestByRefLikeRefReturn()
+        {
+            ByRefLike brl = new ByRefLike();
+            ByRefLike* pBrl = &brl;
+            MethodInfo mi = typeof(TestClass<int>).GetMethod(nameof(TestClass<int>.ByRefLikeRefReturningMethod));
+            try
+            {
+                // Don't use Assert.Throws because that will make a lambda and invalidate the pointer
+                object o = mi.Invoke(null, new object[] { Pointer.Box(pBrl, typeof(ByRefLike*)) });
+                
+                // If this is reached, it means `o` is a boxed byref-like type.
+                Assert.True(false);
```

Comment:
Could you instead throw a XunitException and add a message? So that if the test fails is easier to diagnose?

i.e: 
https://github.com/dotnet/corefx/blob/b996738f877d17b0fe426c75b01f0a8ecdc21e07/src/CoreFx.Private.TestUtilities/src/System/AssertExtensions.cs#L162